src/main/java/org/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,67 @@
+package org.bookstore.controller;
+
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.bookstore.dto.category.CategoryDto;
+import org.bookstore.dto.category.CreateCategoryRequestDto;
+import org.bookstore.model.Book;
+import org.bookstore.model.Category;
+import org.bookstore.service.BookService;
+import org.bookstore.service.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Validated
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/categories")
+public class CategoryController {
+
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PostMapping (value = "")
```

Comment:
```suggestion
    @PostMapping
```
Let’s  fix it in other places